src/client/datascience/notebook/notebookEditorProvider.ts

```diff
@@ -135,7 +138,18 @@ export class NotebookEditorProvider implements INotebookEditorProvider {
         // contents will be ignored
         const cell = new NotebookCellData(NotebookCellKind.Code, '', options?.defaultCellLanguage ?? PYTHON_LANGUAGE);
         const data = new NotebookData([cell]);
+        const edit = new WorkspaceEdit();
         const doc = await this.vscodeNotebook.openNotebookDocument(JupyterNotebookView, data);
+        const metadata = cloneDeep(doc.metadata);
+        if (metadata.language_info !== undefined) {
+            metadata.language_info.name = options?.defaultCellLanguage;
+        } else {
+            metadata.language_info = {
+                name: options?.defaultCellLanguage
+            };
+        }
+        edit.replaceNotebookMetadata(doc.uri, metadata);
```

Comment:
Why not add the metadata to the `data (NotebookDdata)` in line `141`. then we don't need to edit the notebook document.

Reply:
Ahh, I thought `metadata` was frozen. Looks like it's only frozen for `NotebookDocument` not `NotebookData`